src/lib.rs

```diff
@@ -83,6 +83,83 @@ enum Mode {
     WriteDisk { ownership: Ownership },
 }
 
+/// Get all files in a archive using `source` as a reader.
+/// # Example
+///
+/// ```no_run
+/// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+/// use compress_tools::*;
+/// use std::fs::File;
+///
+/// let mut source = File::open("tree.tar")?;
+///
+/// let file_list = list_files(&mut source)?;
+/// # Ok(())
+/// # }
+/// ```
+pub fn get_files<R>(source: R) -> Result<Vec<String>>
+where
+    R: Read,
+{
+    run_with_archive(
+        Mode::AllFormat,
+        source,
+        |archive_reader, _, mut entry| unsafe {
+            let mut list_files = Vec::new();
+            loop {
+                match ffi::archive_read_next_header(archive_reader, &mut entry) {
+                    ffi::ARCHIVE_OK => {
+                        list_files.push(
+                            CStr::from_ptr(ffi::archive_entry_pathname(entry))
+                                .to_str()?
+                                .to_string(),
+                        );
+                    }
+                    _ => return Ok(list_files),
```

Comment:
There are some cases we need to deal with here. From the `libarchive` documentation:

> These functions return ARCHIVE_OK (the operation succeeded), ARCHIVE_WARN (the operation
>  succeeded but a non-critical error was encountered), ARCHIVE_EOF (end-of-archive was
>  encountered), ARCHIVE_RETRY (the operation failed but can be retried), and ARCHIVE_FATAL
>  (there was a fatal error; the archive should be closed immediately).

So we need to handle each of this individually.

Reply:
Sure, you are right i'll change to handle all the cases individually.

Reply:
I was thinking and I believe a better function name would be `list_archive_files`. What do you think?

Reply:
Seems better to me too.